modules/frontend/pipeline/collector_http.go

```diff
@@ -35,29 +39,79 @@ func (r httpCollector) RoundTrip(req *http.Request) (*http.Response, error) {
 		return nil, err
 	}
 
+	err = addNextAsync(ctx, r.consumers, resps, r.combiner, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return r.combiner.HTTPFinal()
+}
+
+func addNextAsync(ctx context.Context, consumers int, resps Responses[combiner.PipelineResponse], c combiner.Combiner, callback func() error) error {
```

Comment:
Nit: I find the function name a bit generic for the pkg

Reply:
agreed. I'm going to change it to`consumeAndCombineResponses`. thoughts?